accumulation_view.php

```diff
@@ -0,0 +1,563 @@
+		echo '"';
+	}
+	foreach ( $listGroups as $groupName )
+	{
+		echo "\n";
+		if ( $hasGroups )
+		{
+			echo '"';
+			$module->echoText( str_replace( '"', '""', $groupName ) );
+			echo '",';
+		}
+		$firstField = true;
+		foreach ( $listColNames as $accStr => $colName )
+		{
+			echo $firstField ? '' : ',';
+			$firstField = false;
+			echo '"';
+			$module->echoText( str_replace( '"', '""',
+			                                formatForOutput( $resultTable[ $accStr ][ $groupName ],
+			                                                 $reportData['display'] ) ) );
+			echo '"';
+		}
+	}
+	if ( $hasGroups && $reportData['group_total'] )
+	{
+		echo "\n";
+		echo '"Total"';
+		foreach ( $listColNames as $accStr => $colName )
+		{
+			echo ',"';
+			$module->echoText( str_replace( '"', '""',
+			                                formatForOutput( $listTotals[ $accStr ],
+			                                                 $reportData['display'] ) ) );
+			echo '"';
+		}
+	}
+	if ( $isInternalRequest )
+	{
+		return;
+	}
+	exit;
+}
+
+
+
+// Handle retrieve report as image.
+if ( isset( $_GET['as_image'] ) && $reportConfig['as_image'] )
+{
+	if ( ! $isInternalRequest )
+	{
+		header( 'Content-Type: image/png' );
+	}
+	// Determine the fonts and character sizes for the report.
+	$imgHeaderFont = 5;
+	$imgDataFont = 4;
+	$imgHeaderCharW = imagefontwidth( $imgHeaderFont );
+	$imgHeaderCharH = imagefontheight( $imgHeaderFont );
+	$imgDataCharW = imagefontwidth( $imgDataFont );
+	$imgDataCharH = imagefontheight( $imgDataFont );
+	$imgHeaderH = $imgHeaderCharH + 2;
+	$imgDataH = $imgDataCharH + 2;
+	// Calculate column widths based on column name string lengths.
+	$imgColumnWidths = [];
+	$imgExtraRows = 0;
+	if ( $hasGroups )
+	{
+		$imgColumnWidths[0] = ( strlen( 'Group' ) * $imgHeaderCharW ) + 5;
+		if ( $reportData['group_total'] )
+		{
+			$thisWidth = ( strlen( 'Total' ) * $imgHeaderCharW ) + 5;
+			if ( $imgColumnWidths[0] < $thisWidth )
+			{
+				$imgColumnWidths[0] = $thisWidth;
+			}
+			$imgExtraRows++;
+		}
+	}
+	foreach ( $listColNames as $accStr => $colName )
+	{
+		$imgColumnWidths[ $accStr ] = ( strlen( $colName ) * $imgHeaderCharW ) + 5;
+	}
+	// Check the data in each column for each record, increase the column widths if necessary.
+	foreach ( $listGroups as $groupName )
+	{
+		if ( $hasGroups )
+		{
+			$thisWidth = ( strlen( $groupName ) * $imgHeaderCharW ) + 5;
+			if ( $imgColumnWidths[0] < $thisWidth )
+			{
+				$imgColumnWidths[0] = $thisWidth;
+			}
+		}
+		foreach ( $listColNames as $accStr => $colName )
+		{
+			$imgParsedData = formatForOutput( $resultTable[ $accStr ][ $groupName ],
+			                                  $reportData['display'] );
+			$thisWidth = ( strlen( $imgParsedData ) * $imgDataCharW ) + 5;
+			if ( $imgColumnWidths[ $accStr ] < $thisWidth )
+			{
+				$imgColumnWidths[ $accStr ] = $thisWidth;
+			}
+		}
+	}
+	// Calculate the image dimensions, create the image, and set the colours (black/white).
```

Comment:
Again - a suggestion for later, not for now, but picking up the current redcap theme colours for text and background rather than forcing black and white would be nice.  Currently the image insert when used looks like it doesn't quite belong.

Reply:
It probably shouldn't always do that, an image could be saved to a file field and then embedded into an email for example. However I may consider options for applying a theme in future.